src/main/java/com/nvidia/spark/rapids/jni/RmmSpark.java

```diff
@@ -23,11 +23,59 @@
 import ai.rapids.cudf.RmmException;
 import ai.rapids.cudf.RmmTrackingResourceAdaptor;
 
+import java.util.Arrays;
+import java.util.Map;
+
 /**
  * Initialize RMM in ways that are specific to Spark.
  */
 public class RmmSpark {
 
+  public enum OomInjectionType {
+    CPU_OR_GPU,
+    CPU,
+    GPU;
+  }
+
+  public static class OomInjection {
+    public final OomInjectionType injectionType;
+    public final int skipCount;
+    public final int oomCount;
+
+    private OomInjection(OomInjectionType injectionType, int skipCount, int oomCount) {
+      this.injectionType = injectionType;
+      this.skipCount = skipCount;
+      this.oomCount = oomCount;
+    }
+
+    public static OomInjection fromString(String s) {
```

Comment:
At a minimum can we have documentation on what this does? "true" and "false" I guess makes since, but why make it all lower case in the plugin instead of making this more lenient on the inputs? Also it would be really nice to document the map config pattern, and what happens if you provide values that are not skip or num_ooms. and what skip/num_ooms is intended to do.